src/persistentstorage.cpp

```diff
@@ -451,15 +451,44 @@ std::string InfluxdbPersistentStorage::read(const std::string& id, const std::st
     return defval;
 }
 
-void InfluxdbPersistentStorage::writelog(const std::string& id, const std::string& value)
+void InfluxdbPersistentStorage::writelog(const std::string &id, const ObjectValue &value)
 {
-    logger_m.debugStream() << "Logging value '" << value << "' for object '" << id << "'" << endlog;
-
+    ObjectValue *objval = (ObjectValue *) &value;
```

Comment:
@fraxinas This cast is not totally wise:
- the parameter is a reference to a const object and this was done on purpose: `writelog` is not expected to mutate `value`, it is only allowed to read it. That's why we don't want to drop the `const` here.
- there is no need to 'degrade' the reference to a pointer. References are just fine in this context (and safer to use)
- the proper way to perform this cast in C++ would be with `&(const_cast<ObjectValue&>(value))`. C-style cast are not considered valid C++ (even though there may be some in this codebase). But again, there is no need to drop the `const`ness of the parameter here.

To determine the actual type of the instance referenced with `value`, you can proceed with `dynamic_cast<const UIntObjectValue&>(value)`. At least, I expect that to work. Let me know otherwise.

Reply:
EDIT: I'm partially wrong for the dynamic_cast suggestion: it is `dynamic_cast<const UIntObjectValue*>(&value)` so that you can check the return value against `NULL` to see if that succeeded. Sorry for the confusion.

Reply:
i ran into problems when trying to run non-const member functions on the const object

for example,
```
        const SwitchingObjectValue val = dynamic_cast<const SwitchingObjectValue*>(&value);
        val.toNumber() ? (influx_value << "t") : (influx_value << "f");
```
results in this error:
`error: passing 'const SwitchingObjectValue' as 'this' argument discards qualifiers [-fpermissive]`

Reply:
@fraxinas OK, got it. That's because `toNumber()` is not qualified as `const`. The API has still a very shallow support for const anyway.
Go ahead with a `const_cast` for this PR and I'll address that in a future PR aimed at const-enabling the rest of the API. Thanks!